tchannel/tornado/dispatch.py

```diff
@@ -233,17 +234,31 @@ def _on_post_response(future):
             e.id = request.id
             connection.send_error(e)
         except Exception as e:
+            exc_type, exc_obj, exc_tb = sys.exc_info()
```

Comment:
This will leave a reference cycle here. Add a try-finally block around the rest of the code and a 'del exc_tb' in the `finally` block.

Related warning in `sys.exc_info()` docs: https://docs.python.org/2/library/sys.html#sys.exc_info